metadoc-js/src/main/scala/metadoc/ScalaTextContentProvider.scala

```diff
@@ -0,0 +1,23 @@
+package metadoc
+
+import scala.scalajs.js.annotation.ScalaJSDefined
+import monaco.IDisposable
+import monaco.Promise
+import monaco.Uri
+import monaco.editor.Editor
+import monaco.editor.IModel
+import monaco.services.ITextModelContentProvider
+import monaco.services.ITextModelService
+
+@ScalaJSDefined
+class ScalaTextContentProvider extends ITextModelContentProvider {
+  override def provideTextContent(resource: Uri): Promise[IModel] = {
+    val model = Editor.createModel(
```

Comment:
One possible fix for the "cannot instantiate second model with the same URI" error would be to first lookup if a model already exists using https://microsoft.github.io/monaco-editor/api/modules/monaco.editor.html#getmodel.